src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -148,10 +148,14 @@ public void visitToken(DetailAST ast) {
             leftCurly = slistToken;
         }
 
-        if (leftCurly != null) {
+        if (leftCurly != null || ast.getType() == TokenTypes.LITERAL_CASE) {
```

Comment:
@romani Also since the `slist` to be analysed for detecting empty `case` token is its next sibling and not a child (like I've already mentioned) and hence whenever a `case` token is being checked the program would have to work a bit differently hence there needs to be a little bit of "separate processing". (In fact I think there's also similar separate processing invovled for an empty `switch`, look at line 158)